### PR TITLE
Feature/add cybermetrics

### LINKED ISSUE #403: [Cybermetrics] Addition of logging with Firebase
**Is your feature request related to a problem? Please describe.**
We lack a lot of information about app usage to make meaningful change to accommodate most users default settings. It could be great to log information in firebase about events happening.

**Describe the solution you'd like**
The solution could have the following informations logged through the function `LogEvent()`:
- reordering, deletion or reinitialization of dashboard card;
- tab profile/grade in student page;
- each quicklinks click (we could see most to least popular);
- Each navigation tab;
- Each click on our social network link in dashboard applets card.

This list could of course be extended if other information would be useful. 😄 

**Describe alternatives you've considered**
N/A

**Additional context**
All logs should have a description that fits their unique purpose. For example, we should see in firebase:

`Accessing navbar element: Student page`

Instead of:

`Accessing navbar element`

### PR DESCRIPTION
**Pull request by camillebrulotte**
### ⁉️ Related Issue
Closes #403 

## 📖 Description
Added analytics to : 
- Reordering, deletion and reinitialization of dashboard cards;
- Each click on our social network link in dashboard page;
- Tab Notes/Profil;
- Add test for quicklinks analytic;
- BottomBar navigation;
- More;
- Schedule buttons;

Also, reorder programs by status asked by @MysticFragilist (Actif first, diplomé, dossier fermé, ...)

### CODE REVIEW

**Review by MysticFragilist — APPROVED** [review]
LGTM! :)

**Review by MysticFragilist — APPROVED** [review]
LGTM!

File: lib/ui/views/dashboard_view.dart [hunk 1]
```diff
@@ -47,6 +48,8 @@ class _DashboardViewState extends State<DashboardView>
     with TickerProviderStateMixin {
   Text progressBarText;
   final NavigationService _navigationService = locator<NavigationService>();
+  final AnalyticsService _analyticsService = locator<AnalyticsService>();
+  static const String tag = "DashboardViewModel";
```
Comment (eneiss) [inline lib/ui/views/dashboard_view.dart:52]: Ca devrait pas être DashboardView (au lieu de ViewModel ?)

**Review by JonathanDuvalV — APPROVED** [review]

File: lib/ui/views/dashboard_view.dart (hunk 1 above)
Comment (camillebrulotte) [reply lib/ui/views/dashboard_view.dart:52]: Bien vue ! :D

**Review by MysticFragilist — APPROVED** [review]

**Review by JonathanDuvalV — APPROVED** [review]

**Pull request merged**